rfcs/button-component.md

```diff
@@ -0,0 +1,81 @@
+# Button Component
+
+Rough DOM composition:
+
+```
++-------------------------------------------------------------------+
+| BaseButton (display: flex; justifyContent: space-between)         |
+| +-----------------+-----------------------+---------------------+ |
+| |  startEnhancer  |         label         |     endEnhancer     | |
+| |  display:flex   |                       |     display:flex    | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| +-----------------+-----------------------+---------------------+ |
++-------------------------------------------------------------------+
+```
+
+### Exports
+
+* `Button`
+* `StyledButton`
+* `StyledButtonLabel`
+* `StyledStartEnhancer`
+* `StyledEndEnhancer`
+* `KIND`
+* `SIZE`
+* `SHAPE`
+
+### <Button/> API
+
+* `label: node | function` - Optional
+  A label in the button.
+* `startEnhancer: node | function` - Optional
+  A helper rendered at the start of the button.
+* `endEnhancer: node | function` - Optional
+  A helper rendered at the end of the button.
+* `disabled: boolean` - Optional
+  Defines if the button disabled.
```

Comment:
We should probably also make sure that people can pass through `type="submit"` to their buttons.

Actually it looks like there are a lot of other form/a11y related props people may need to pass through: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

Maybe we should just pass through `...rest` props to the root element so users can pass whatever accessibility / aria attributes they might need?

Reply:
I agree to pass `...rest`. Additionally I imagine it's not uncommon for users to change the `StyledButton` to a link using `$as="a"` (Styletron's `$as`) and pass in `href` and `target`.